jpegenc/models/rgb2ycbcr/rgb2ycbcr.py

```diff
@@ -0,0 +1,144 @@
+#!/bin/python
+from myhdl import *
+from commons import *
+from math import *
+
+from math import *
+
+#Fixed point representation with 1 sign bit and 14 fractional bits
+fract_bits = 14
+sign_bits=1
+
+Y_COEFF=[0.2999,0.587,0.114]
+CB_COEFF=[-0.1687,-0.3313,0.5]
+CR_COEFF=[0.5,-0.4187,-0.0813]
+OFFSET=[0,128,128]
+
+Y=[int(round(Y_COEFF[i]*(2**fract_bits )))for i in range(3)]
+Cb=[int(round(CB_COEFF[i]*(2**fract_bits ))) for i in range(3)]
+Cr=[int(round(CR_COEFF[i]*(2**fract_bits ))) for i in range(3)]
+Offset=[int(round(OFFSET[i]*(2**fract_bits ))) for i in range(3)]
+
+class RGB(object):
+
+    def __init__(self, nbits=8):
+        self.nbits=nbits
+        self.red = Signal(intbv(0)[nbits:])
+        self.green = Signal(intbv(0)[nbits:])
+        self.blue = Signal(intbv(0)[nbits:])
+
+    def next(self, r, g, b):
+        self.red.next = r
+        self.green.next = g
+        self.blue.next = b
+
+    def bitLength(self): return self.nbits
+
+
+class YCbCr(object):
+
+    def __init__(self, nbits=8):
+        self.nbits = nbits
+        self.y = Signal(intbv(0)[nbits:])
+        self.cb = Signal(intbv(0)[nbits:])
+        self.cr = Signal(intbv(0)[nbits:])
+
+    def bitLength(self): return self.nbits
```

Comment:
As mentioned in other conversations, these classes can be refactored into a separate file.  As the project grows other modules will use these classes (this refactoring can occur in the future - post this PR).